examples/serial/board/qemu_arm_virt/serial.system

```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<system>
+    <memory_region name="uart" size="0x1_000" phys_addr="0x9000000" />
+
+    <memory_region name="tx_data_driver" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="tx_data_client" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="rx_data_driver" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="rx_data_client" size="0x200_000" page_size="0x200_000" />
+
+    <memory_region name="tx_data_client2" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="rx_data_client2" size="0x200_000" page_size="0x200_000" />
+
+    <!-- shared memory for ring buffer mechanism : serial -->
+    <memory_region name="rx_free_serial_driver" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="rx_active_serial_driver" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="tx_free_serial_driver" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="tx_active_serial_driver" size="0x200_000" page_size="0x200_000"/>
+
+    <memory_region name="tx_free_serial_client" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="tx_active_serial_client" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="rx_free_serial_client" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="rx_active_serial_client" size="0x200_000" page_size="0x200_000"/>
+
+    <memory_region name="tx_free_serial_client2" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="tx_active_serial_client2" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="rx_free_serial_client2" size="0x200_000" page_size="0x200_000"/>
+    <memory_region name="rx_active_serial_client2" size="0x200_000" page_size="0x200_000"/>
+
+    <protection_domain name="uart" priority="100" pp="true">
+        <program_image path="uart.elf" />
+
+        <map mr="uart" vaddr="0x5_000_000" perms="rw" cached="false" setvar_vaddr="uart_base" />
+
+        <!-- shared memory for ring buffer mechanism -->
+        <map mr="rx_free_serial_driver" vaddr="0x4_000_000" perms="rw" cached="true" setvar_vaddr="rx_free" />
+        <map mr="rx_active_serial_driver" vaddr="0x4_200_000" perms="rw" cached="true" setvar_vaddr="rx_active" />
+        <map mr="tx_free_serial_driver" vaddr="0x4_400_000" perms="rw" cached="true" setvar_vaddr="tx_free" />
+        <map mr="tx_active_serial_driver" vaddr="0x4_600_000" perms="rw" cached="true" setvar_vaddr="tx_active" />
+
+        <map mr="tx_data_driver" vaddr="0x2_200_000" perms="rw" cached="true" />
+        <map mr="rx_data_driver" vaddr="0x2_600_000" perms="rw" cached="true" />
+
+        <!-- UART interrupt -->
+        <irq irq="33" id="1" />
```

Comment:
This IRQ is for the QEMU.  Can it be parameterised here to use the 255,edge for Meson or 33.level for QEMU?

Reply:
So you can use the same system.xml file for both boards?

Reply:
I don't want to do any extra indirection on the examples, it makes sense for LionsOS but not for these isolated examples.